protocol/kafka_confluent/v2/message.go

```diff
@@ -0,0 +1,143 @@
+/*
+ Copyright 2023 The CloudEvents Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package kafka_confluent
+
+import (
+	"bytes"
+	"context"
+	"strconv"
+	"strings"
+
+	"github.com/cloudevents/sdk-go/v2/binding"
+	"github.com/cloudevents/sdk-go/v2/binding/format"
+	"github.com/cloudevents/sdk-go/v2/binding/spec"
+	"github.com/confluentinc/confluent-kafka-go/v2/kafka"
+)
+
+const (
+	prefix         = "ce-"
+	contentTypeKey = "Content-Type"
+)
+
+const (
+	KafkaOffsetKey    = "kafkaoffset"
+	KafkaPartitionKey = "kafkapartition"
+	KafkaTopicKey     = "kafkatopic"
+	KafkaMessageKey   = "kafkamessagekey"
+)
+
+var specs = spec.WithPrefix(prefix)
+
+// Message represents a Kafka message.
+// This message *can* be read several times safely
+type Message struct {
+	internal   *kafka.Message
+	properties map[string][]byte
+	format     format.Format
+	version    spec.Version
+}
+
+// Check if Message implements binding.Message
+var (
+	_ binding.Message               = (*Message)(nil)
+	_ binding.MessageMetadataReader = (*Message)(nil)
+)
+
+func NewMessage(msg *kafka.Message) *Message {
+	var contentType, contentVersion string
+	properties := make(map[string][]byte, len(msg.Headers)+3)
+	for _, header := range msg.Headers {
+		k := strings.ToLower(string(header.Key))
+		if k == strings.ToLower(contentTypeKey) {
+			contentType = string(header.Value)
+		}
+		if k == specs.PrefixedSpecVersionName() {
+			contentVersion = string(header.Value)
```

Comment:
question: what if `header.Value` is `1.0.2` but MUST be `1.0` according to 

> Compliant event producers MUST use a value of 1.0 when referring to this version of the specification
https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/spec.md#specversion

Reply:
I‘m not sure how to handle the specversion with value 1.0.2. Maybe just validate the 'major' and 'minor' versions whether compliant with "1.0"? and if not, then print a warning message?

I checked the [other protocol](https://github.com/cloudevents/sdk-go/blob/0a3095dbbd1db5c6bbd6a039c927dcac876d4f1b/protocol/kafka_sarama/v2/message.go#L74). Then all just assert whether the specversion key exists and ignore the value. 

Do you have any other advice?

Reply:
Yeah, IMHO this is a shortcoming in the other implementations. I'm fine leaving this here as is then given it's consistent with our other implementations. cc/ @duglin

Reply:
I like being forgiving and just ignoring the patch version number - if we want to do anything different at all. I'm ok with the current choice of only looking for 0.3 and 1.0.